databuilder/query_engines/sqlite.py

```diff
@@ -0,0 +1,159 @@
+import contextlib
+from functools import cache, cached_property, singledispatchmethod
+
+import sqlalchemy
+from sqlalchemy.dialects.sqlite.pysqlite import SQLiteDialect_pysqlite
+from sqlalchemy.sql import operators
+
+from databuilder.query_model import Function, SelectColumn, SelectPatientTable, Value
+from databuilder.sqlalchemy_utils import get_referenced_tables
+
+from .base import BaseQueryEngine
+
+
+class SQLiteQueryEngine(BaseQueryEngine):
+
+    sqlalchemy_dialect = SQLiteDialect_pysqlite
+
+    def get_query(self, variable_definitions):
+        variable_expressions = {
+            name: self.get_sql(definition)
+            for name, definition in variable_definitions.items()
+        }
+        population_expression = variable_expressions.pop("population")
+        return self.get_results_query(variable_expressions, population_expression)
+
+    def get_results_query(self, variable_expressions, population_expression):
+        # Get all referenced tables
+        tables = set(get_tables(population_expression))
+        for expression in variable_expressions.values():
+            tables.update(get_tables(expression))
+
+        assert len(tables) > 0, "No tables found in query"
+
+        patient_table = self.get_patient_table(tables)
+        other_tables = tables - {patient_table}
```

Comment:
This can't be part of that set, can it?

Reply:
Not under the current implementation. But if we changed that to return a vanilla table like `Patients` then it could, and I thought maybe it made the intention clearer.

Reply:
Up to you.